Source/Core/Common/FileUtil.cpp

```diff
@@ -276,33 +279,68 @@ bool DeleteDir(const std::string& filename)
   return false;
 }
 
+// Repeatedly invokes func until it returns true or max_attempts failures.
+// Waits after each failure, with each delay doubling in length.
+template <typename FuncType>
+bool AttemptMaxTimesWithExponentialDelay(const FuncType& func, int max_attempts,
```

Comment:
```suggestion
static bool AttemptMaxTimesWithExponentialDelay(const FuncType& func, int max_attempts,
```

since this is only used in this translation unit